packages/nimble-components/src/dialog/index.ts

```diff
@@ -125,11 +124,33 @@ export class Dialog<CloseReason = void> extends FoundationElement {
         if (this.preventDismiss) {
             event.preventDefault();
         } else {
-            this.resolveShow!(UserDismissed);
-            this.resolveShow = undefined;
+            this.notifyClosed(UserDismissed);
         }
         return true;
     }
+
+    /**
+     * @internal
+     */
+    public closeHandler(): void {
+        if (this.resolveShow) {
+            // If
+            // - the browser implements dialogs with the CloseWatcher API, and
+            // - the user presses ESC without first interacting with the dialog (e.g. clicking, scrolling),
+            // the cancel event is not fired and the dialog just closes.
+            this.notifyClosed(UserDismissed);
+        }
+    }
+
+    private notifyClosed(reason: CloseReason | UserDismissed): void {
```

Comment:
I don't think notify is the right term. The convention is notify is used for event listeners or observation.

The action is to resolve the promise, maybe `completeResolveShow` or `finishResolveShow` or `doResolveShow`

Reply:
Went with `doResolveShow`